### PR TITLE
Option to set max count for how many messages to purge

### LINKED ISSUE #295: Get message API allows unlimited amount of messages to be acked/rejected
There needs to be a limit since trying to reject hundreds of thousands of messages will take up lots of resources if they are to be returned from the API.

* Strict limit on number of messages returned
* Ability to reject without returning the messages?

### PR DESCRIPTION
**Pull request by snichme**
Allows the user to purge X number of messages instead of usign "Get message" feature to remove the
from the queue which is slow if there are a lot of messages to be dropped.

Fixes #295

### CODE REVIEW

**Comment by snichme** [comment]
image
Looks weird when the buttons isn't aligned, any suggestions?

**Comment by jage** [comment]
> Looks weird when the buttons isn't aligned, any suggestions?

Tried to tweak a bit, but didn't find any quick-fixes. Also noticed that in a slightly less wide window the button text line wraps. Slightly better if we switch pause/purge/delete to `cols-4`. 

Maybe this is good enough for now, and then we can do a design-refactoring later on?

**Comment by snichme** [comment]
> Maybe this is good enough for now, and then we can do a design-refactoring later on?

Sounds good to me, should have a bigger overhaul on the responsive design as I've seen other places where it doesn't keep up

**Review by jage — APPROVED** [review]

File: src/avalanchemq/queue/queue.cr [hunk 1]
```diff
@@ -843,11 +843,19 @@ module AvalancheMQ
...
-      @log.debug { "Purged #{count} messages" }
+    def purge(max_count : Int? = nil) : UInt32
+      @log.info "Purging"
+      delete_count = if max_count.nil?
+                       @ready.purge
+                     else
+                       Math.min(max_count, @ready.size)
+                         .times
+                         .compact_map { |_| @ready.shift? }
+                         .size
```
[the remark this answers is not in the document]
Comment (jage) [reply src/avalanchemq/queue/queue.cr:854]: Great catch, will fix!

**Pull request merged**